test/Core.Test/Services/GroupServiceTests.cs

```diff
@@ -30,12 +33,116 @@ public GroupServiceTests()
             );
         }
 
-        // Remove this test when we add actual tests. It only proves that
-        // we've properly constructed the system under test.
         [Fact]
-        public void ServiceExists()
+        public async Task SaveAsync_DefaultGroupId_CreatesGroupInRepository()
         {
-            Assert.NotNull(_sut);
+            var organization = new Models.Table.Organization
+            {
+                Id = Guid.NewGuid(),
+                UseGroups = true,
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            await _sut.SaveAsync(group);
+
+            await _groupRepository.Received().CreateAsync(group);
```

Comment:
Please validate event creation as well -- it's a mostly enterprise feature, but this kind of auditing is super important.